samples/basic_app_bar.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: page
+catalog: basic_app_bar
+title: "AppBar Basics"
+
+permalink: /catalog/basic_app_bar/
+---
+
+
+![iOS AppBar](/images/samples/ios_animated_list_small.png){.catalog}
+![Android AppBar](/images/samples/animated_list_small.png){.catalog}
+
+
+### Description
+An AppBar with a title, actions, and an overflow dropdown menu.
+One of the app's choices can be selected with an action button or the menu.
+
+An app that displays one of a half dozen choices with an icon and a title.
+The two most common choices are available as action buttons and the remaining
+choices are included in the overflow dropdow menu.
+
+### Related
+See also:
+- The "Layout-Structure" section of the material design specification:
+    <https://material.io/guidelines/layout/structure.html#structure-app-bar>
+
+### Sample App
+```dart
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+// This app is a stateful, it tracks the user's current choice.
+class BasicAppBarSample extends StatefulWidget {
+  @override
+  _BasicAppBarSampleState createState() => new _BasicAppBarSampleState();
+}
+
+class _BasicAppBarSampleState extends State<BasicAppBarSample> {
+  Choice _selectedChoice = choices[0]; // The app's "state".
+
+  void _select(Choice choice) {
+    setState(() { // Causes the app to rebuild with the new _selectedChoice.
+      _selectedChoice = choice;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      home: new Scaffold(
+        appBar: new AppBar(
+          title: const Text('Basic AppBar'),
+          actions: <Widget>[
+            new IconButton( // action button
+              icon: new Icon(choices[0].icon),
+              onPressed: () { _select(choices[0]); },
+            ),
+            new IconButton( // action button
+              icon: new Icon(choices[1].icon),
+              onPressed: () { _select(choices[1]); },
+            ),
+            new PopupMenuButton<Choice>( // overflow menu
+              onSelected: _select,
+              itemBuilder: (BuildContext context) {
+                return choices.skip(2).map((Choice choice) {
+                  return new PopupMenuItem<Choice>(
+                    value: choice,
+                    child: new Text(choice.title),
+                  );
+                }).toList();
+              },
+            ),
+          ],
+        ),
+        body: new Padding(
+          padding: const EdgeInsets.all(16.0),
+          child: new ChoiceCard(choice: _selectedChoice),
+        ),
+      ),
+    );
+  }
+}
+
+class Choice {
```

Comment:
How can we make it ultra clear which code is "optional" (specific to this sample), and which code is required (will always need) ?

Do I need to make a Choice class when making an AppBar?